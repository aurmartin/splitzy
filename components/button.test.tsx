import Button from "@/components/button";
import { render, screen, userEvent } from "@testing-library/react-native";
import * as React from "react";

it("renders correctly", () => {
  render(<Button>Press me!</Button>);
  expect(screen.toJSON()).toMatchSnapshot();
});

it("renders correctly with type", () => {
  render(<Button type="primary">Press me!</Button>);
  expect(screen.toJSON()).toMatchSnapshot();
});

it("renders correctly with style", () => {
  render(<Button style={{ backgroundColor: "red" }}>Press me!</Button>);
  expect(screen.toJSON()).toMatchSnapshot();
});

it("handles onPress events", async () => {
  const user = userEvent.setup();

  const onPressMock = jest.fn();
  render(<Button onPress={onPressMock}>Press me!</Button>);

  await user.press(screen.getByText("Press me!"));
  expect(onPressMock).toHaveBeenCalled();
});

it("renders disabled state correctly", () => {
  const onPressMock = jest.fn();
  render(
    <Button disabled onPress={onPressMock}>
      Press me!
    </Button>,
  );
  expect(screen.toJSON()).toMatchSnapshot();
});
