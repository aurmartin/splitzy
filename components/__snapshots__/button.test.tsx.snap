// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      [
        {
          "alignItems": "center",
          "borderRadius": 40,
          "height": 40,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        },
        {
          "backgroundColor": "hsl(200 40% 50%)",
        },
        undefined,
      ],
      {
        "overflow": "hidden",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Press me!
  </Text>
</View>
`;

exports[`renders correctly with style 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      [
        {
          "alignItems": "center",
          "borderRadius": 40,
          "height": 40,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        },
        {
          "backgroundColor": "hsl(200 40% 50%)",
        },
        {
          "backgroundColor": "red",
        },
      ],
      {
        "overflow": "hidden",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Press me!
  </Text>
</View>
`;

exports[`renders correctly with type 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      [
        {
          "alignItems": "center",
          "borderRadius": 40,
          "height": 40,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        },
        {
          "backgroundColor": "hsl(200 40% 50%)",
        },
        undefined,
      ],
      {
        "overflow": "hidden",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Press me!
  </Text>
</View>
`;

exports[`renders disabled state correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      [
        {
          "alignItems": "center",
          "borderRadius": 40,
          "height": 40,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        },
        {
          "backgroundColor": "hsl(200 40% 50%)",
        },
        undefined,
      ],
      {
        "overflow": "hidden",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        {
          "color": "white",
        },
      ]
    }
  >
    Press me!
  </Text>
</View>
`;
